WordPress/gutenberg
# Add onClick prop to FormFileUpload

## Description

Fixes #39267

This PR adds a `onInputFileClick` prop on the `FormFileUpload`.

Currently when there's an error on the `FormFileUpload` and you need to upload the same file again, it doesn't emit the `onChange` event and the user get stuck until choose another file.

With this change, we can add `<FormFileUpload onInputFileChange={ (e) => e.target.value }>` on the parent component, this way the `<input type="file">` inside `FormFileUpload` will empty its file after click, allowing to upload the same file again.

I didn't make it runs by default for compatibility purposes.

## Testing Instructions




1. Go to a page that has the `FormFileUpload` component with a `onChange` event.
2. Add `onInputFileChange={ (e) => e.target.value }` to the component.
3. Choose any file.
4. Choose the same file again, it should emit the `onChange` event again.

## Screenshots 
![Untitled](https://user-images.githubusercontent.com/1693223/157125582-911c2f94-1458-4724-b09b-9caf19f9207f.gif)

Current behavior, the correct would be show the loading again and re-upload the file again, but as the hidden `<input type="file">` doesn't change its value, it doesn't emit the `onChange` event, this PR fixes that.

## Types of changes





## Checklist:
